sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -169,26 +171,32 @@ class InMemoryTable(
 
     new WriteBuilder with SupportsTruncate with SupportsOverwrite with SupportsDynamicOverwrite {
       private var writer: BatchWrite = Append
+      private var streamingWriter: StreamingWrite = StreamingAppend
 
       override def truncate(): WriteBuilder = {
         assert(writer == Append)
         writer = TruncateAndAppend
+        streamingWriter = StreamingTruncateAndAppend
         this
       }
 
       override def overwrite(filters: Array[Filter]): WriteBuilder = {
         assert(writer == Append)
         writer = new Overwrite(filters)
+        // streaming writer doesn't have equivalent semantic
```

Comment:
Hmm, does this mean for streaming case, we won't reach here?

Reply:
Yes, at least for now. If we would like to be sure, we may be able to assign dummy one and throw error on calling `buildForStreaming()`. Probably it'd be much clearer.

Reply:
Just changed to explicitly fail for the case.